modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/ratings/RatingsManagerImpl.kt

```diff
@@ -0,0 +1,37 @@
+package au.com.shiftyjelly.pocketcasts.repositories.ratings
+
+import au.com.shiftyjelly.pocketcasts.models.db.AppDatabase
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastRatings
+import au.com.shiftyjelly.pocketcasts.servers.podcast.PodcastCacheServerManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+class RatingsManagerImpl @Inject constructor(
+    private val cacheServerManager: PodcastCacheServerManager,
+    appDatabase: AppDatabase,
+) : RatingsManager, CoroutineScope {
+    private val podcastRatingsDao = appDatabase.podcastRatingsDao()
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    override fun observePodcastRatings(podcastUuid: String) =
+        podcastRatingsDao.observePodcastRatings(podcastUuid)
+            .map { it.firstOrNull() ?: PodcastRatings(podcastUuid, 0.0) }
```

Comment:
I return a blank rating if there's no rating in the DB for the podcast so that UI is still rendered (without any filled stars). This is subject to change based on final UI decision. It doesn't look like the best way to do it, let me know if you have suggestions. 

Also, it returns a flowable so that it can fit in the [loadPodcasts(...)](https://github.com/Automattic/pocket-casts-android/blob/main/modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastViewModel.kt#L92) transformations, where podcast rating it loaded after the podcast is loaded.

Reply:
Good idea! I applied that in ca1119e